### PR TITLE
Now showing temp, and rider pads as charts under the map.

### PR DESCRIPTION
**Pull request by ebabel**
Pretty neat, but not 100% sure I like the UX. @kwatkins you or your lady have any ideas on a cleaner UI/UX?

### CODE REVIEW

**Comment by ebabel** [comment]
screen shot 2017-11-30 at 12 34 14 am
screen shot 2017-11-30 at 12 34 57 am

**Comment by kwatkins** [comment]
Nice, I'll ask!

**Comment by kwatkins** [comment]
Asked, she gave the Nike running app as an example of a possible UI. Running apps are about the splits and elevation, which really doesn't carry over to OW stats, so it's more about what features we want to show, i.e tracks line color matched to a stat

Def. something we should all think about/discuss, some suggestions or things that would be cool to see, 
- Parts that were tough or I ate it, where the board flipped and pads were off 
- Battery life over the ride, maybe tie that in to the GPS track line (green to red) 

based on this Linda can mock the ui and any graphics we need

**Comment by ebabel** [comment]
ya, def need to add the crash/slip events to the map! should we have a
toggle to show/hide them?

I'll install nike running app, good idea.

colored line makes sense, I saw a talk at lyft about how to do this.

probably also want a chart that shows battery, spent amps, regen amps as
three lines or maybe like a fill chart would be nice(and challenging to get
it to look good)

what did you think about blocking the user from scrolling while dragging
their finger on the chart? I'm not a fan...but couldn't think of a nicer
way to do it without a ton of work. maybe nike has this covered?

File: app/src/main/java/net/kwatts/powtools/RideDetailActivity.java [hunk 1]
```diff
@@ -112,21 +122,38 @@ private void retrieveData(ArrayList<Entry> timeSpeedMap) {
...
                     timeLocationMap.put(time, new LatLng(moment.getGpsLatDouble(), moment.getGpsLongDouble()));
                 }
-                Attribute attribute = database.attributeDao().getFromMomentAndKey(moment.id, "speed");
-                if (attribute != null && attribute.getValue() != null) {
-                    String value = attribute.getValue();
-                    timeSpeedMap.add(new Entry(time, Float.valueOf(value)));
+//                Attribute attribute = database.attributeDao().getFromMomentAndKey(moment.id, "speed");
+                List<Attribute> attributes = database.attributeDao().getFromMoment(moment.id);
+                for (Attribute attribute : attributes) {
+                    if (attribute != null
```
Comment (twyatt) [inline app/src/main/java/net/kwatts/powtools/RideDetailActivity.java]: This `null` check needed? If so, then probably better just to make sure and never save `null`s to the db.

File: app/src/main/java/net/kwatts/powtools/RideDetailActivity.java [hunk 2]
```diff
@@ -400,4 +422,222 @@ public void onNothingSelected() {
...
+        // enable touch gestures
+        lineChart.setTouchEnabled(true);
+        lineChart.setOnChartGestureListener(new LockScrollbarGestureListener(scrollView));
+        lineChart.setOnChartValueSelectedListener(new OnChartValueSelectedListener() {
+            @Override
+            public void onValueSelected(Entry entry, Highlight h) {
+                Long entryX = (long) entry.getX();
+                if (timeLocationMap.containsKey(entryX)) {
+                    for (Marker mapMarker : mapMarkers) {
+                        mapMarker.remove();
```
Comment (twyatt) [inline app/src/main/java/net/kwatts/powtools/RideDetailActivity.java]: Also remove item from `Set`?

File: app/src/main/java/net/kwatts/powtools/RideDetailActivity.java [hunk 3]
```diff
@@ -400,4 +422,222 @@ public void onNothingSelected() {
...
+                    }
+                    LatLng latLng = timeLocationMap.get(entryX);
+                    mapMarkers.add(googleMap.addMarker(new MarkerOptions().position(latLng)));
+                }
+            }
+
+            @Override
+            public void onNothingSelected() {
+                for (Marker mapMarker : mapMarkers) {
+                    mapMarker.remove();
```
Comment (twyatt) [inline app/src/main/java/net/kwatts/powtools/RideDetailActivity.java]: Also remove from `Set`?
`mapMarkers.clear()` after loop?

File: app/src/main/java/net/kwatts/powtools/RideDetailActivity.java [hunk 4]
```diff
@@ -400,4 +422,222 @@ public void onNothingSelected() {
...
+        // enable touch gestures
+        lineChart.setTouchEnabled(true);
+        lineChart.setOnChartGestureListener(new LockScrollbarGestureListener(scrollView));
+        lineChart.setOnChartValueSelectedListener(new OnChartValueSelectedListener() {
+            @Override
+            public void onValueSelected(Entry entry, Highlight h) {
+                Long entryX = (long) entry.getX();
+                if (timeLocationMap.containsKey(entryX)) {
+                    for (Marker mapMarker : mapMarkers) {
+                        mapMarker.remove();
```
Comment (twyatt) [inline app/src/main/java/net/kwatts/powtools/RideDetailActivity.java]: Also remove from `Set`?

File: app/src/main/java/net/kwatts/powtools/RideDetailActivity.java [hunk 5]
```diff
@@ -400,4 +422,222 @@ public void onNothingSelected() {
...
+                    }
+                    LatLng latLng = timeLocationMap.get(entryX);
+                    mapMarkers.add(googleMap.addMarker(new MarkerOptions().position(latLng)));
+                }
+            }
+
+            @Override
+            public void onNothingSelected() {
+                for (Marker mapMarker : mapMarkers) {
+                    mapMarker.remove();
```
Comment (twyatt) [inline app/src/main/java/net/kwatts/powtools/RideDetailActivity.java]: Also remove from `Set`?

File: app/src/main/java/net/kwatts/powtools/components/LockableScrollView.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            default:
+                return super.onTouchEvent(ev);
+        }
+    }
+
+    @Override
+    public boolean onInterceptTouchEvent(MotionEvent ev) {
+        // Don't do anything with intercepted touch events if
+        // we are not scrollable
+        if (!mScrollable) return false;
```
Comment (twyatt) [inline app/src/main/java/net/kwatts/powtools/components/LockableScrollView.java]: nit: Hungarian notation (granted from copied code, but still)

File: app/src/main/java/net/kwatts/powtools/components/LockableScrollView.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        else return super.onInterceptTouchEvent(ev);
```
Comment (twyatt) [inline app/src/main/java/net/kwatts/powtools/components/LockableScrollView.java]: Curly braces would make this more consistent with the codebase and easier to read.

**Comment by kwatkins** [comment]
Merging, know UI and all of that can be cleaner but having this in at least moves it forward.

**Pull request merged**